packages/functional-tests/pages/login.ts

```diff
@@ -276,7 +280,7 @@ export class LoginPage extends BaseLayout {
     return this.page.fill(selectors.RECOVERY_KEY_TEXT_INPUT, code);
   }
 
-  async loginHeader() {
+  async isUserLoggedIn() {
```

Comment:
Is there a reason not to use the `waitForSelector` here as well? Seems like doing this could clean up a number of expect statements in our tests.

Reply:
Yes, in a lot of tests I have used `innerText()`, `isVisible` or other functions to work off of this `isUserLoggedIn()` header. Hence I didn't change it to use `waitForSelector()`. I have changed all the other headers though.

Reply:
You are right. But somehow just for this selector the using` waitFor()` together with a `isVisible()` has proven to work better. I think that's because loading the settings page after a successful login takes a bit of time and depending on a state of a selector rather than on a specified(default) timeout works better for this one. Its also suggested in the docs.
<img width="894" alt="Screen Shot 2023-08-01 at 12 37 57 PM" src="https://github.com/mozilla/fxa/assets/62558650/31cc2119-63f3-4dd4-9eb0-578cc186bb74">
<img width="885" alt="Screen Shot 2023-08-01 at 12 38 52 PM" src="https://github.com/mozilla/fxa/assets/62558650/2cfce48d-8e03-4c53-92ee-d4ad285b0675">

When I ran the tests with `waitForSelector()` alot of tests failed since it was unable to load the page and the test quit abruptly.
<img width="1349" alt="Screen Shot 2023-08-01 at 12 44 38 PM" src="https://github.com/mozilla/fxa/assets/62558650/f4e0d1bd-62d9-4923-8d23-9f3a8d441106">

But with waitFor() and isVisible(), all the tests passed

<img width="1364" alt="Screen Shot 2023-08-01 at 12 44 02 PM" src="https://github.com/mozilla/fxa/assets/62558650/267cb33c-0338-42e7-96b5-21b45fe998a3">